### PR TITLE
Expect explicit resource in capability

### PR DESCRIPTION
**Pull request by leplatrem**
This illustrates the behaviour I would have expected in the exposed capabilties.

Suppose we have this config:
```ini
kinto.signer.group_check_enabled = false
kinto.signer.to_review_enabled = false

kinto.signer.resources =
    /buckets/dev -> /buckets/stage -> /buckets/prod

kinto.signer.dev.normandy.to_review_enabled = true
kinto.signer.dev.normandy.group_check_enabled = true
```

I would expect the following capabilites:
```js

{
    "to_review_enabled": False,
    "group_check_enabled": False,
    "resources": {
                "source": {"bucket": "dev",
                                "collection": None},
                "preview": {"bucket": "preview",
                            "collection": None},
                "destination": {"bucket": "prod",
                                "collection": None},
                "reviewers_group": "stage-{collection_id}-reviewers",
            }, {
                "source": {"bucket": "dev",
                           "collection": "normandy"},
                "preview": {"bucket": "preview",
                            "collection": "normandy"},
                "destination": {"bucket": "prod",
                                "collection": "normandy"},
                "reviewers_group": "dev-normandy-reviewers",
                "to_review_enabled": True,
                "group_check_enabled": True,
            }]
}
```

### CODE REVIEW

**Comment by leplatrem** [comment]
Note: The branch is based on #281

File: kinto_signer/__init__.py [hunk 1]
```diff
@@ -34,6 +35,36 @@ def includeme(config):
         raise ConfigurationError(error_msg)
     resources = utils.parse_resources(raw_resources)
 
+    # Expand the resources with the ones that come from per-bucket resources
+    # and have specific settings.
+    # For example, resource is ``/buckets/dev -> /buckets/prod``
```
Comment (glasserc) [inline kinto_signer/__init__.py]: Nit: Maybe something like `consider the case when`?

File: kinto_signer/__init__.py [hunk 2]
```diff
@@ -34,6 +35,36 @@ def includeme(config):
...
+    # and there is a setting ``signer.dev.recipes.signer_backend = foo``
+    for key, resource in resources.items():
+        # If collection is not None, there is nothing to expand :)
+        if resource['source']['collection'] is not None:
+            continue
+        bid = resource['source']['bucket']
+        # Match setting names like signer.stage.specific.autograph.hawk_id
+        matches = [(k, re.search(r'signer\.{0}\.([^\.]+)\.(.+)'.format(bid), k))
+                   for k, v in settings.items()]
```
Comment (glasserc) [inline kinto_signer/__init__.py]: Could we keep `v` instead of `k` here?

File: kinto_signer/__init__.py [hunk 3]
```diff
@@ -34,6 +35,36 @@ def includeme(config):
...
+        found = [(k, m.group(1), m.group(2)) for (k, m) in matches if m]
+        if len(found) == 0:
+            # No specific collection setting for this bucket
+            continue
```
Comment (glasserc) [inline kinto_signer/__init__.py]: I don't really think we need an explicit continue if the rest of the code is in the for loop.

**Comment by leplatrem** [comment]
I rebased and squashed commits to ease review. 

@glasserc could you have a final overlook please?

**Pull request merged**